### PR TITLE
x-set-installed: Allow more options

### PR DESCRIPTION
**Pull request by autoantwort**
This would be really useful for testing ports

### CODE REVIEW

File: src/vcpkg/commands.setinstalled.cpp [hunk 1]
```diff
@@ -20,11 +20,19 @@ namespace vcpkg::Commands::SetInstalled
     static constexpr StringLiteral OPTION_ONLY_DOWNLOADS = "only-downloads";
     static constexpr StringLiteral OPTION_WRITE_PACKAGES_CONFIG = "x-write-nuget-packages-config";
     static constexpr StringLiteral OPTION_NO_PRINT_USAGE = "no-print-usage";
+    static constexpr StringLiteral OPTION_USE_HEAD_VERSION = "head";
+    static constexpr StringLiteral OPTION_EDITABLE = "editable";
```
Comment (BillyONeal) [inline src/vcpkg/commands.setinstalled.cpp:24]: I'm somewhat opposed to these given what x-set-installed is intended to do. We can't 'make the world be exactly this list' if we are `--head`.

I'm all for `--enforce-port-checks` and `--allow-unsupported` though.

Comment (autoantwort) [reply src/vcpkg/commands.setinstalled.cpp:24]: I have already expected that this is controversial. I personally don't see a conflict between "make the world be exactly this list" and head mode. Then it is "make the world be exactly this list with head builds".

**Review by BillyONeal — CHANGES_REQUESTED** [review]
@JavierMatosD @ras0219-msft @dan-shaw @markle11m @AugP and I discussed these cases today:

* Nobody objects to `allow-unsupported` or `enforce-port-checks`
* We would like some form of spec or end to end scenario to describe how `--head` or `--editable` would be used as proposed. Notably, to do what `x-set-installed` says on the tin, every `--head` thing would need to be reinstalled, and all edits made under `--editable` would be lost, so it isn't clear when the former applies or what effect the latter has. For example, the right thing might be to add a `test-port` command or a `--rebuild` or similar to `install` instead which more closely aligns with what you're trying to do.

**Comment by ras0219-msft** [comment]
Moving to draft for now, given above.

**Pull request closed without merging**